contractcourt/htlc_outgoing_contest_resolver_test.go

```diff
@@ -0,0 +1,158 @@
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet"
+)
+
+const (
+	outgoingContestHtlcExpiry = 110
+)
+
+// TestHtlcOutgoingResolverTimeout tests resolution of an offered htlc that
+// timed out.
+func TestHtlcOutgoingResolverTimeout(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	ctx := newOutgoingResolverTestContext(t)
+	ctx.resolve()
+	ctx.notifyEpoch(outgoingContestHtlcExpiry - 1)
+	ctx.waitForResult(true)
+}
+
+// TestHtlcOutgoingResolverRemoteClaim tests resolution of an offered htlc that
+// is claimed by the remote party.
+func TestHtlcOutgoingResolverRemoteClaim(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	ctx := newOutgoingResolverTestContext(t)
+	ctx.resolve()
+	preimage := lntypes.Preimage{}
+	ctx.notifier.spendChan <- &chainntnfs.SpendDetail{
+		SpendingTx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					Witness: [][]byte{
+						{0}, {1}, {2}, preimage[:],
+					},
+				},
+			},
+		},
+	}
+	<-ctx.preimageDB.newPreimages
+	ctx.waitForResult(false)
+}
+
+type outgoingResolverTestContext struct {
+	resolver     *htlcOutgoingContestResolver
+	notifier     *mockNotifier
+	preimageDB   *mockWitnessBeacon
+	resolveErr   chan error
+	nextResolver ContractResolver
+	t            *testing.T
+}
+
+func newOutgoingResolverTestContext(t *testing.T) *outgoingResolverTestContext {
+	notifier := &mockNotifier{
+		epochChan: make(chan *chainntnfs.BlockEpoch),
+		spendChan: make(chan *chainntnfs.SpendDetail),
+		confChan:  make(chan *chainntnfs.TxConfirmation),
+	}
+
+	checkPointChan := make(chan struct{}, 1)
+	resolutionChan := make(chan ResolutionMsg, 1)
+
+	preimageDB := newMockWitnessBeacon()
+
+	chainCfg := ChannelArbitratorConfig{
+		ChainArbitratorConfig: ChainArbitratorConfig{
+			Notifier:   notifier,
+			PreimageDB: preimageDB,
+			DeliverResolutionMsg: func(msgs ...ResolutionMsg) error {
+				if len(msgs) != 1 {
+					return fmt.Errorf("expected 1 "+
+						"resolution msg, instead got %v",
+						len(msgs))
+				}
+
+				resolutionChan <- msgs[0]
+				return nil
+			},
+		},
+	}
+
+	outgoingRes := lnwallet.OutgoingHtlcResolution{
+		Expiry: outgoingContestHtlcExpiry,
+		SweepSignDesc: input.SignDescriptor{
+			Output: &wire.TxOut{},
+		},
+	}
+
+	resolver := &htlcOutgoingContestResolver{
+		htlcTimeoutResolver: htlcTimeoutResolver{
+			ResolverKit: ResolverKit{
+				ChannelArbitratorConfig: chainCfg,
+				Checkpoint: func(_ ContractResolver) error {
+					checkPointChan <- struct{}{}
+					return nil
+				},
+			},
+			htlcResolution: outgoingRes,
+		},
+	}
+
+	return &outgoingResolverTestContext{
+		resolver:   resolver,
+		notifier:   notifier,
+		preimageDB: preimageDB,
+		t:          t,
+	}
+}
+
+func (i *outgoingResolverTestContext) resolve() {
+	// Start resolver.
+	i.resolveErr = make(chan error, 1)
+	go func() {
+		var err error
+		i.nextResolver, err = i.resolver.Resolve()
```

Comment:
seems like it would be better to define a channel where both the resolver and error could be sent, to avoid this awkward struct field.

Reply:
infinitely better, done.